lib/src/source/git.dart

```diff
@@ -40,6 +41,49 @@ class GitSource extends CachedSource {
     });
   }
 
+  /// Gets the list of all versions from git tags.
+  /// Anything that Version.parse understands is considered a version,
+  /// it will also attempt to strip off a preceding 'v' e.g. v1.0.0
+  Future<List<Pubspec>> getVersions(String name, description) async {
+    if (!_useVersionTags(description)) {
+      // No need to get versions if the useVersionTags option is false
+      return super.getVersions(name, description);
+    }
+
+    PackageId id =
+        new PackageId(name, this.name, null, _getDescription(description));
+    String cachePath = _repoCachePath(id);
+    if (!entryExists(cachePath)) {
+      // Must have the repo cloned in order to list its tags
+      await _clone(_getUrl(id), cachePath);
+    }
+
+    List results = await git.run(["tag", "-l"], workingDir: cachePath);
+    List<Pubspec> validVersions = [];
+    for (String version in results) {
```

Comment:
This would require an additional loop - the `where` which selects the valid version strings, and another that loops through the valid version strings and awaits the pubspec for each.

I sort of prefer it as one loop, but if you would rather have that separation I can change it.